### PR TITLE
Add Rspec Tests to Application

### LINKED ISSUE #25: Add Tests to Application
I believe we should include this from the ground up, before the application turns into a 100 pound gorilla.

### PR DESCRIPTION
**Pull request by uzorjchibuzor**
Resolves #25

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -0,0 +1,2 @@
+--require spec_helper
```
Comment (datae95) [inline .rspec:1]: when you `--require rails_helper` here, you can omit it in every test file

File: spec/models/link_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+require "rails_helper"
```
Comment (datae95) [inline spec/models/link_spec.rb:1]: See above

File: spec/models/link_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+RSpec.describe Link, :type => :model do
+  describe "useful links that helps with rails development" do
+    context "with valid attributes" do 
+      let(:link) {FactoryBot.create(:link)}
+      it "creates a valid link" do
+        expect(link).to be_valid
```
Comment (datae95) [inline spec/models/link_spec.rb:8]: Here the only thing you test is whether FactoryBot works

**Review by datae95 — DISMISSED** [review]

File: .rspec (hunk 1 above)
Comment (datae95) [reply .rspec:1]: My typical `.rspec` file is 
```
--require rails_helper
--format documentation
--color
--order defined
```

Comment (uzorjchibuzor) [reply .rspec:1]: > My typical `.rspec` file is
> 
> ```
> --require rails_helper
> --format documentation
> --color
> --order defined
> ```

I knew I was forgetting something

Comment (datae95) [reply .rspec:1]: > I knew I was forgetting something

You weren't forgetting something. It's a matter of what you want or don't want. Especially `--order defined` is purely taste. I like to know when which test is executed, for others it is more important that the order of the tests is randomized, so they can be sure, that the green test does not rely on previously run tests.

File: spec/models/link_spec.rb (hunk 2 above)
Comment (uzorjchibuzor) [reply spec/models/link_spec.rb:1]: > See above

That's gone in the incoming commit

File: .rspec (hunk 1 above)
Comment (uzorjchibuzor) [reply .rspec]: I like it to, makes the output easy to read.

**Comment by uzorjchibuzor** [comment]
I saw a review notification in the mail but can't find it
Screen Shot 2022-06-29 at 6 14 27 PM
 here @datae95

**Comment by uzorjchibuzor** [comment]
> I trust you with this one :) Testing is just on horizon for me

I always had to go back and read documentation sometimes too. But when you see those tests passing one after the other, it gives you confidence that your application is healthy at least.

**Pull request merged**